app/services/feeds/import.rb

```diff
@@ -122,7 +122,7 @@ def create_articles_from_user_feed(user, feed)
       articles = []
 
       feed.entries.reverse_each do |item|
-        next if medium_reply?(item) || article_exists?(user, item)
+        next if Feeds::CheckItemMediumReply.call(item) || Feeds::CheckItemPreviouslyImported.call(item, user)
```

Comment:
These two services only seem to be used together, in a guard clause. I'm starting to wonder if we should tie them together with another service, e.g. `Feeds::ShouldProcessItem` or something? It could be really simple for now:

```ruby
module Feeds
  class ShouldProcessItem
    def self.call(item, user)
      CheckItemMediumReply.call(item) && CheckItemPreviouslyImported.call(item, user)
    end
  end
end
```

Then the guard clauses need to switch:

```ruby
next unless Feed::ShouldProcessItem.call(item, user)
return unless Feed::ShouldProcessItem.call(item, user)
```

If we ever need more checks they wouldn't need to be added in two different places.

Reply:
Good call, though I'm not 100% convinced on adding a third service that's just a wrapper for two. It'd be nice to use dependency injection, so something like:

```
def call(item, user, checkers: [array of services to send .call on])
  # something like checkers.all? { |c| c.call?(item, user)
end
```

but they both feel a bit over-engineered for now

Reply:
Yes, fair point, I'm not saying we should do either. I guess something here just rubs me the wrong way, e.g. if they are only used together, maybe this split was premature and both of them together should have moved into a single validity checker service.